### ISSUE TITLE
Can't select search result after enabling location services

### ISSUE THREAD

**Issue by ddogfoodd** [organicmaps/organicmaps#9825]
**Describe the issue**
I have searched but haven't found another issue describing this problem.
More often than not I cannot select a search result. The search correctly shows the results and I can tap the view on map button which works, however when I tap on any of the search results nothing happens.

EDIT: This happens after enabling Location Services via the android quick settings menu.

**Steps to reproduce**
1. Turn on Location in Android Quick Settings
2. Go to Search (click on magnifying glass icon)
3. Search for anything
4. Tap on any of the results
5. Sometimes nothing happens

**Expected behaviour**
I expect the search result that is tapped to be shown on map.

**Screenshots**
I can upload a recording if that helps. There really isn't much to show, but if you need it, let me know.

**System information:**
 - Operating system and its version: Android 14
 - Organic Maps version: 2024.11.27-12-Web
 - Device Model: Fairphone 5

**Comment by map-per** [comment]
I can't reproduce this, the search works on my phone

**Comment by pastk** [comment]
Do you have the phone unlocked when you do it?
Or maybe you have OM on your lockscreen when it happens?

**Comment by RedAuburn** [comment]
> Is there a way to spoof the location, so I don't leak my precise location? Else I will need to edit the video later.

You can use https://f-droid.org/en/packages/cl.coders.faketraveler/

make sure to set it as the mock location app in developer options:
> Requirements: In order to work, you need to allow Fake Traveler to mock locations. You have to [enable Developer options](https://developer.android.com/studio/debug/dev-options?hl=en-419) and select this app in Settings/System/Developer options/Select mock location app option.

**Comment by RedAuburn** [comment]
> screen-20241207-024227.1.mp4

Thanks! That's _very_ odd.
(nice wallpaper!)

**Comment by biodranik** [comment]
@ddogfoodd do bookmarks work in a similar dialog? You may need to add a few first to test.

And to clarify: if the location is disabled, everything works fine, correct?

**Comment by biodranik** [comment]
@ddogfoodd do you see Google Location Services setting on OM options? If yes, can you please try to change its value, restart OM, and try to reproduce the issue again?

**Comment by biodranik** [comment]
Is your phone un-googled? Do you have microg installed? Is the networking GPS provider set up/enabled in the system settings?

**Comment by felix-knopp** [comment]
I have the same issue and it makes organic maps unusable for me. If I wait (about one to two minutes) the search result becomes clickable.


For me this seems not related to activating location services. It just happens all the time. Rarely it just works as expected but I don't see a pattern. 


 https://github.com/user-attachments/assets/a16ce1ab-321c-4b33-a351-00b8cbced389

**Comment by ddogfoodd** [comment]
I have since switched to CalyxOS with MicroG, the error still occurs.

**Comment by biodranik** [comment]
@felix-knopp what is your device, Android version, system firmware? Do you have microg or are you using Google Location services? Any other hints or patterns on how to reproduce it?

Does pressing the system's back button and repeating the search help?

> I have since switched to CalyxOS with MicroG, the error still occurs.

@ddogfoodd your device and pattern is still the same, right? You enable location => it stops working?

**Comment by ddogfoodd** [comment]
> Does pressing the system's back button and repeating the search help?

no, if I repeat the search, the issue persists for the new seach

> [@ddogfoodd](https://github.com/ddogfoodd) your device and pattern is still the same, right? You enable location => it stops working?

yes, same as before. works as expected without location services, when enabling them it stops working

**Comment by zloidemon** [comment]
> [@felix-knopp](https://github.com/felix-knopp) what is your device, Android version, system firmware? Do you have microg or are you using Google Location services? Any other hints or patterns on how to reproduce it?
> 

[Alexander Borsuk](https://github.com/biodranik) Could you kindly review the issue description first, as it may already contain the necessary details about the device?

**Comment by felix-knopp** [comment]
> [@felix-knopp](https://github.com/felix-knopp) what is your device, Android version, system firmware? Do you have microg or are you using Google Location services? Any other hints or patterns on how to reproduce it?
> 
> Does pressing the system's back button and repeating the search help?
> 
> > I have since switched to CalyxOS with MicroG, the error still occurs.
> 
> [@ddogfoodd](https://github.com/ddogfoodd) your device and pattern is still the same, right? You enable location => it stops working?

Hi, sorry for the late reply. I totally missed this.

I'm on a Fairphone 5 (FP5), just updated to Android 15, Build Nr. FP5.VT28.C.042.20250623
I'm using the stock Android version (with Google Location service).
The problem is still there but sporadic and seems to be happening less often though. 
I have not noticed any pattern on when the problem occurs, yet.

**Comment by sschwarzer** [comment]
I have the same problem on my Fairphone 5 (original firmware, latest firmware update) with enabled location services. Most searches don't work, but currently they do work (also several times in a row), without changing the location settings.

Edit: I should be more specific for later reference. My current firmware is Android 15, FP5.VT28.C.042.20250623, but I'm very sure I had this issue also before with Android 14.

**Comment by biodranik** [comment]
So, we have 3 users with the same issue, all with Fairphone 5. Are there similar issues mentioned in the Fairphone bug issue tracker?

**Comment by sschwarzer** [comment]
Good point. I web-searched for some time (30 minutes?) and didn't find anything.

I tried

fp5 taps not registering

with variations fp5 -> fairphone 5 and/or adding "location" to the search terms.

Only when I include organicmaps in the search, I find something relevant - this Github issue. ;-) Relatedly, Organicmaps is the only app where I've run into the issue of taps not registering.

**Comment by biodranik** [comment]
If any of you can enable developer mode, and connect your devices to the computer via cable, and then debug the layout using Android Studio, that would be great! https://developer.android.com/studio/debug/layout-inspector

Maybe there is some kind of "invisible" window on top of Organic Maps that catches all touches and doesn't pass them through to OM?

Another option is to check logs via `adb logcat -v color` after connecting over USB, and see what's in logs when touches do not work. Maybe there will be some warnings or errors.

**Comment by ddogfoodd** [comment]
@biodranik thank you for the guidance! That sounds easy enough.
However, now that I want to debug, I can't seem to reproduce it anymore.
Maybe the others can test if it works for them as well on OrganicMaps version `2025.03.02-7-Web`. If the error re-occurs, I can debug it.

I'd say we wait a bit for the others to test, then close. We could still re-open if necessary.

**Comment by sschwarzer** [comment]
> Another option is to check logs via adb logcat -v color after connecting over USB, and see what's in logs when touches do not work. Maybe there will be some warnings or errors.

I've never done any mobile software development, but I managed to generate the attached log file with `adb logcat -v color`. I removed some lines for privacy reasons, e.g. regarding other apps, but I hope everything relevant is still there.

I also wanted to repeat the search with a better grasp of my actions vs. the timestamps in the logfile, but curiously I couldn't reproduce the problem anymore (even without any changes in the location settings, which seemed to help in another case above).

[organicmaps_section.log.gz](https://github.com/user-attachments/files/22559742/organicmaps_section.log.gz)

**Comment by ddogfoodd** [comment]
Thank you for testing @sschwarzer!
Might be fixed then. Before, it happened very frequently so I guess this rules out luck. I'd be fine with closing until another person comes across this error.

**Comment by sschwarzer** [comment]
> Thank you for testing [@sschwarzer](https://github.com/sschwarzer)! Might be fixed then. Before, it happened very frequently so I guess this rules out luck. I'd be fine with closing until another person comes across this error.

Oh, it seems we have a misunderstanding. The bug still happens for me.

My problem with the log was that I wanted to provide a better description in my ticket comment about which log entries were from which user interface actions, but when I tried _again_ (for the planned second log), the bug didn't occur. But it occurred when I recorded the log that I attached above.

**Comment by ddogfoodd** [comment]
Oh yeah my bad for misunderstanding. In that case I will also try again.

**Comment by ddogfoodd** [comment]
> If any of you can enable developer mode, and connect your devices to the computer via cable, and then debug the layout using Android Studio, that would be great! https://developer.android.com/studio/debug/layout-inspector

I can't seem to get the Android Studio Layout Inspector to work. I can connect and mirror my device but the Component Tree in the Layout Inspector says: "Application not inspectable. Deploy or navigate to a debuggable app to begin inspection."

Do I have to install and use a debug version of the app or am I missing some setting or something?
I have already enabled "Enable view attribute inspection" on my device.

**Comment by biodranik** [comment]
@ddogfoodd you need a debug build. Any pull request in this repo builds debug apks and attaches them as zip archives. You can try to debug any of it (they can be installed separately from the main app). Check one of the runs here, for example: https://github.com/organicmaps/organicmaps/actions/runs/18626078824

**Comment by ddogfoodd** [comment]
While the bug seems to occur rarely for me now, I can still reliably reproduce it by:
1. Enable Android location services
2. Start OMaps App
3. Tap the locate button
4. Search for something
5. Try to tap on a result
6. If the result is not clickable -> done. Else go back by swipe gesture and repeat steps 3 to 5.

This way I compared the Layout Inspection Trees of the non-clickable/bugged result page and the clickable/non-bugged result page.
(I used the Debug build from the run linked in the comment above.)
However, I couldn't make out a difference. There seems to be the same amount of layout elements unless I have overseen something. Also, in the bugged case, when the result finally becomes tappable after the layout tree doesn't seem to change.

Maybe there is a difference more deeply hidden in some layout element's values, but I don't feel experienced enough to guess where I should look.

**Comment by biodranik** [comment]
Thanks for testing and sharing your results. Would changing any other location/location+network-related system settings help? OM should be restarted after each experiment.
